### PR TITLE
Build APKs without a local copy of the engine sources

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 1]
```diff
@@ -53,6 +53,10 @@ enum ArtifactType {
   snapshot,
   shell,
   mojo,
+  android_classes_dex,
+  android_icudtl_dat,
+  android_keystore,
+  android_libsky_shell
```
Comment (abarth) [inline packages/flutter_tools/lib/src/artifacts.dart]: The usual style for Dart enums is camelCase

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 2]
```diff
@@ -122,6 +126,30 @@ class ArtifactStore {
       type: ArtifactType.mojo,
       targetPlatform: TargetPlatform.linux
     ),
+    const Artifact._(
+      name: 'classes.dex',
```
Comment (abarth) [inline packages/flutter_tools/lib/src/artifacts.dart]: The name is meant to be a human-understandable name... "Compiled Java code" ?

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 3]
```diff
@@ -122,6 +126,30 @@ class ArtifactStore {
...
+      fileName: 'classes.dex',
+      type: ArtifactType.android_classes_dex,
+      targetPlatform: TargetPlatform.android
+    ),
+    const Artifact._(
+      name: 'icudtl.dat',
```
Comment (abarth) [inline packages/flutter_tools/lib/src/artifacts.dart]: "ICU data table" ?

File: packages/flutter_tools/lib/src/artifacts.dart [hunk 4]
```diff
@@ -122,6 +126,30 @@ class ArtifactStore {
...
+      targetPlatform: TargetPlatform.android
+    ),
+    const Artifact._(
+      name: 'keystore',
+      fileName: 'chromium-debug.keystore',
+      type: ArtifactType.android_keystore,
+      targetPlatform: TargetPlatform.android
+    ),
+    const Artifact._(
+      name: 'libsky_shell',
```
Comment (abarth) [inline packages/flutter_tools/lib/src/artifacts.dart]: Compiled C++ code ?

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/lib/src/artifacts.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/artifacts.dart]: done

File: packages/flutter_tools/lib/src/artifacts.dart (hunk 2 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/artifacts.dart]: done

File: packages/flutter_tools/lib/src/artifacts.dart (hunk 3 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/artifacts.dart]: done

File: packages/flutter_tools/lib/src/artifacts.dart (hunk 4 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/artifacts.dart]: done

**Pull request merged**